libyul/backends/evm/EVMDialect.cpp

```diff
@@ -140,18 +140,19 @@ map<YulString, BuiltinFunctionForEVM> createBuiltins(langutil::EVMVersion _evmVe
 				_assembly.appendAssemblySize();
 			else
 			{
+				auto [isData, subIdPath] = _context.currentObject->pathToSubObject(dataName.str());
 				yulAssert(
-					_context.subIDs.count(dataName) != 0,
+					!subIdPath.empty() | isData,
 					"Could not find assembly object <" + dataName.str() + ">."
 				);
-				_assembly.appendDataSize(_context.subIDs.at(dataName));
+				_assembly.appendDataSize(isData ? std::vector<size_t>{_context.subIDs[dataName]} : subIdPath);
```

Comment:
```suggestion
				_assembly.appendDataSize(isData ? vector<size_t>{_context.subIDs[dataName]} : subIdPath);
```
There's also one more below.